sdk/openai/openai/test/public/openai.spec.ts

```diff
@@ -2,36 +2,43 @@
 // Licensed under the MIT license.
 
 import { Recorder } from "@azure-tools/test-recorder";
-import { assert } from "@azure/test-utils";
+import { assert, isNode, matrix } from "@azure/test-utils";
 import { Context } from "mocha";
 import { OpenAIClient } from "../../src/OpenAIClient.js";
-import { createClient, startRecorder } from "./utils/recordedClient.js";
+import { AuthMethod, createClient, startRecorder } from "./utils/recordedClient.js";
 
-describe("client", () => {
-  let recorder: Recorder;
-  let client: OpenAIClient;
+matrix([["AzureAPIKey", "AAD", "OpenAIKey"]] as const, async function (authMethod: AuthMethod) {
+  describe(`[${authMethod}] Client`, () => {
+    let recorder: Recorder;
+    let client: OpenAIClient;
 
-  beforeEach(async function (this: Context) {
-    recorder = new Recorder(this.currentTest);
-    recorder = await startRecorder(this.currentTest);
-    client = createClient("AzureAPIKey", { recorder });
-  });
+    beforeEach(async function (this: Context) {
+      if (!isNode && authMethod == "AAD") {
+        this.skip();
+      }
```

Comment:
There is currently an issue with AAD and the browser that we're still investigating, so skipping those tests for now.